packages/sdk/src/evm/core/classes/smart-wallet-factory.ts

```diff
@@ -0,0 +1,187 @@
+
+export class SmartWalletFactory<TContract extends IAccountFactory> implements DetectableFeature {
+
+  featureName = FEATURE_SMART_WALLET_FACTORY.name;
+  private contractWrapper: ContractWrapper<IAccountFactory>;
+
+  // utilities
+  public events: ContractEvents<IAccountFactory>;
+
+  constructor(
+    contractWrapper: ContractWrapper<TContract>,
+  ) {
+    this.contractWrapper = contractWrapper;
+    
+    this.events = new ContractEvents(this.contractWrapper);
+  }
+
+  getAddress(): string {
+    return this.contractWrapper.readContract.address;
+  }
+
+	/*********************************
+   * READ FUNCTIONS
+   *******************************/
+
+	/**
+   * Get the deterministic address of the smart wallet that will be created for the given admin.
+   * 
+   * @example
+   * ```javascript
+   * const walletAddress = await contract.smartWalletFactory.predictWalletAddress(admin);
+   * ```
+   * @param admin - The admin of the smart wallet.
+   * @param extraData - (Optional) Extra data to be passed to the smart wallet on creation.
+   * @returns the deterministic address of the smart wallet that will be created for the given admin.
+   * 
+   * @twfeature SmartWalletFactory
+   */
+	public async predictWalletAddress(admin: string, extraData?: BytesLike): Promise<string> {
+		let data: BytesLike = ethers.utils.toUtf8Bytes("");
+		if(extraData) {
+			data = extraData;
+		}
+		return this.contractWrapper.readContract.getAddress(admin, data);
+	}
+
+	/**
+   * Get all signers who have authority on the given smart wallet.
+   * 
+   * @example
+   * ```javascript
+   * const allSigners = await contract.smartWalletFactory.getAssociatedSigners(admin);
+   * ```
+   * @param wallet - The smart wallet address.
+   * @returns all signers who have authority on the given smart wallet.
+   * 
+   * @twfeature SmartWalletFactory
+   */
+	public async getAssociatedSigners(wallet: string): Promise<string[]> {
+		return this.contractWrapper.readContract.getSignersOfAccount(wallet);
+	}
+
+	/**
+   * Get all wallets on which the given signer has authority.
+   * 
+   * @example
+   * ```javascript
+   * const allWallets = await contract.smartWalletFactory.getAssociatedWallets(admin);
+   * ```
+   * @param signer - The smart wallet address.
+   * @returns all wallets on which the given signer has authority.
+   * 
+   * @twfeature SmartWalletFactory
+   */
+	public async getAssociatedWallets(signer: string): Promise<string[]> {
+		return this.contractWrapper.readContract.getAccountsOfSigner(signer);
+	}
+
+  /**
+   * Get all wallets created via the smart wallet factory.
+   * 
+   * @example
+   * ```javascript
+   * const allWallets = await contract.smartWalletFactory.getAllWallets();
+   * ```
+   * 
+   * @returns all wallets created via the smart wallet factory.
+   * 
+   * @twfeature SmartWalletFactory
+   */
+  public async getAllWallets(): Promise<AccountEvent[]> {
+    const filter = {
+      fromBlock: 0,
+      toBlock: "latest",
+    }
```

Comment:
this very well may fail in RPCs but we can try

Reply:
I thought we could use the ethers [`deployTransaction.blockNumber`](https://docs.ethers.org/v5/api/contract/contract/#Contract--properties) api to set the `fromBlock` but that wouldn't work due to this requirement:

```
If the Contract object is the result of a ContractFactory deployment, this is the transaction which was used to deploy the contract.
```

Reply:
@adam-maj how far are we from being able to plug specific events into our RPC / dedicated endpoint pulling data from chainsaw you think?